cli/azd/main.go

```diff
@@ -55,7 +55,10 @@ func main() {
 	// print a warning if we are not. Note that we don't print this warning when the CLI version
 	// is exactly 0.0.0-dev.0, which is a sentinel value used for `internal.Version` when
 	// a version is not explicitly applied at build time (i.e. dev builds installed with `go install`)
-	if ok {
+	//
+	// Don't write this message when JSON output is enabled, since in that case we use stderr to return structured
+	// information about command progress.
+	if !isJsonOutput() && ok {
```

Comment:
I wonder if this is a sign we should start moving this down into the main execution of the command, instead of `main.go` to be able to reference parsed flags. It would also allow the error check to only happen when valid flags are passed to `azd`.

Reply:
Agreed.  I'll start thinking about how to do this.